Packs/1Password/Integrations/OnePasswordEventCollector/OnePasswordEventCollector_test.py

```diff
@@ -0,0 +1,323 @@
+    params = {
+        'audit_events_limit': expected_audit_events_limit,
+        'sign_in_attempts_limit': expected_sign_in_attempts_limit,
+    }
+
+    audit_events_limit = get_limit_param(params, event_type='audit events')
+    sign_in_attempts_limit = get_limit_param(params, event_type='sign in attempts')
+
+    assert audit_events_limit == int(expected_audit_events_limit)
+    assert sign_in_attempts_limit == int(expected_sign_in_attempts_limit)
+
+
+def test_get_unauthorized_event_types():
+    """
+    Given:
+        - JSON response from 'Auth introspect' endpoint and a list of configured event types.
+
+    When:
+        - Calling get_unauthorized_event_types.
+
+    Assert:
+        - Ensure correct list of unauthorized event types.
+    """
+    from OnePasswordEventCollector import get_unauthorized_event_types
+
+    event_types = ['audit events', 'item usage actions', 'sign in attempts']
+    mock_response = util_load_json('test_data/introspection_response.json')
+    unauthorized_event_types = get_unauthorized_event_types(mock_response, event_types)
+    assert unauthorized_event_types == []
+
+
+def test_add_fields_event():
+    """
+    Given:
+        - A raw 1Password event of type 'audit event'.
+
+    When:
+        - Calling add_fields_to_event.
+
+    Assert:
+        - Ensure the '_time' and 'event_type' fields are added and correctly set.
+    """
+    from OnePasswordEventCollector import add_fields_to_event, arg_to_datetime, DATE_FORMAT
+
+    event_timestamp = '2024-12-02T11:54:19.710457472Z'
+    expected_event_time = arg_to_datetime(event_timestamp).strftime(DATE_FORMAT)
+
+    event_type = 'audit event'
+    raw_event = {
+        'uuid': '12345',
+        'timestamp': event_timestamp,
+        'action': 'create',
+        'object_type': 'device',
+    }
+    add_fields_to_event(raw_event, event_type)
+
+    assert raw_event['_time'] == expected_event_time
+    assert raw_event['event_type'] == event_type
+
+
+def test_client_get_events_valid_inputs(authenticated_client: Client, requests_mock: RequestsMock):
+    """
+    Given:
+        - A OnePasswordEventCollector.Client instance with valid inputs to the get_events method.
+
+    When:
+        - Calling Client.get_events.
+
+    Assert:
+        - Ensure no exception is raised and the raw API response is as expected.
+    """
+    from OnePasswordEventCollector import urljoin
+
+    event_type = 'Sign in attempts'
+    from_date = datetime(2024, 12, 2, 11, 50)
+
+    event_url = urljoin(BASE_URL, '/signinattempts')
+    mock_response = util_load_json('test_data/signinattempts_response.json')
+    requests_mock.post(event_url, json=mock_response)
+
+    with assert_not_raises(Exception):
+        response = authenticated_client.get_events(event_type=event_type, from_date=from_date)
+
+    assert response == mock_response
+
+
+@pytest.mark.parametrize(
+    'event_type, error_message',
+    [
+        pytest.param(
+            'Random event',
```

Comment:
```suggestion
            'some unsupported event type',
```
easier to understand than `random`, imo

Reply:
Updated